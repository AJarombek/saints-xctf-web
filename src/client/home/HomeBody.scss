/*
 * Sass styling for the HomeBody component
 * @author Andrew Jarombek
 * @since 1/19/2020
 */

@import "../fonts";

.sxctf-home-body {

  > div {
    height: 120px;
    margin: 250px 0 450px 5%;

    @media screen and (max-width: 1600px) {
      margin: 200px 0 400px 5%;
    }

    @media screen and (max-width: 1400px) {
      margin: 200px 0 380px 5%;
    }

    @media screen and (max-width: 1350px) {
      margin: 200px 0 350px 5%;
    }

    @media screen and (max-width: 1300px) {
      margin: 200px 0 300px 5%;
    }

    @media screen and (max-width: 1210px) {
      margin: 200px 0 250px 5%;
    }

    @media screen and (max-width: 1120px) {
      margin: 180px 0 240px 5%;
    }

    h2 {
      @include roboto-bold;
      color: #bbb;
      width: 60%;
      font-size: 36px;
    }

    .options {
      display: flex;
    }

    .aj-text-button {
      button {
        color: #eee !important;
      }

      &:hover, &:active {
        button {
          color: white !important;
        }
      }
    }
  }
}
