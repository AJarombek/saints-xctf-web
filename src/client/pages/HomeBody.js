/**
 * Home body containing action buttons.
 * @author Andrew Jarombek
 * @since 1/12/2020
 */

import {AJButton} from 'jarombek-react-components';

const HomeBody = () => {
    return (
        <div className="sxctf-home-body">
            <AJButton type="contained">Try the Demo</AJButton>
            <AJButton type="text">Get Started</AJButton>
        </div>
    );
};

export default HomeBody;
