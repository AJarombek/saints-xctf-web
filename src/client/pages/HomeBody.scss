/*
 * Sass styling for the HomeBody component
 * @author Andrew Jarombek
 * @since 1/19/2020
 */

@import "../fonts";

.sxctf-home-body {
  height: 100vh;

  > div {
    margin: calc(100px + 15vh) 0 0 5%;

    h2 {
      @include roboto-bold;
      color: #bbb;
      width: 60%;
      font-size: 36px;
    }

    .options {
      display: flex;
    }

    .aj-text-button {
      button {
        color: #eee !important;
      }

      &:hover, &:active {
        button {
          color: white !important;
        }
      }
    }
  }
}
