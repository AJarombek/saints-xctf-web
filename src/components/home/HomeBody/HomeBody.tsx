/**
 * Home body containing action buttons.
 * @author Andrew Jarombek
 * @since 1/12/2020
 */

import React from 'react';
import { AJButton } from 'jarombek-react-components';
import { FeatureFlag } from '../../shared/FeatureFlag';

const HomeBody: React.FunctionComponent = () => {
  return (
    <div className="sxctf-home-body">
      <div>
        <h2>Cross Country and Track & Field Team Exercise Logs</h2>
        <div className="options">
          <FeatureFlag flag="DEMO">
            <AJButton type="contained">Try the Demo</AJButton>
          </FeatureFlag>
          <AJButton type="text">Get Started</AJButton>
        </div>
      </div>
    </div>
  );
};

export default HomeBody;
