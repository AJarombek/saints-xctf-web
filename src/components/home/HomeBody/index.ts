import HomeBody from "./HomeBody";

export default HomeBody;
