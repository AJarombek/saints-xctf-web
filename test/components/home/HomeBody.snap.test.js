/**
 * Snapshot test for the HomeBody component.
 * @author Andrew Jarombek
 * @since 2/20/2020
 */

import React from 'react';
import HomeBody from '../../../src/components/home/HomeBody';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <HomeBody />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
