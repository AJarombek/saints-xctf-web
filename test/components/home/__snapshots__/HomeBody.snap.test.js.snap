// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sxctf-home-body"
>
  <div>
    <h2>
      Cross Country and Track & Field Team Exercise Logs
    </h2>
    <div
      className="options"
    >
      <div
        className="ajBaseButton-0-2-4 ajBaseButtonEnabled-0-2-5 aj-text-button ajTextButton-0-2-1 ajTextButtonEnabled-0-2-2"
        onClick={[Function]}
      >
        <button
          disabled={false}
          type="button"
        >
          Get Started
        </button>
      </div>
    </div>
  </div>
</div>
`;
